api/v1alpha2/agentpool_types.go

```diff
@@ -73,14 +73,29 @@ type AgentDeploymentAutoscaling struct {
 	//+optional
 	//+kubebuilder:default:=300
 	CooldownPeriodSeconds *int32 `json:"cooldownPeriodSeconds,omitempty"`
+
+	// CoolDownPeriod configures the period to wait between scaling up and scaling down
+	//+optional
+	CooldownPeriod *AgentDeploymentAutoscalingCooldownPeriod `json:"cooldownPeriod,omitempty"`
+}
+
+// AgentDeploymentAutoscalingCooldownPeriod configures the period to wait between scaling up and scaling down
+type AgentDeploymentAutoscalingCooldownPeriod struct {
+	// ScaleUpSeconds is the time to wait before scaling up.
+	//+optional
+	ScaleUpSeconds *int32 `json:"scaleUpSeconds,omitempty"`
+
+	// ScaleDownSeconds is the time to wait before scaling down.
+	//+optional
+	ScaleDownSeconds *int32 `json:"scaleDownSeconds,omitempty"`
```

Comment:
To preserve the existing API, I have left these as optional with no default value and default to using the value from `cooldownPeriodSeconds` when not set. Later, we can deprecate the `cooldownPeriodSeconds` field and default these fields to a specific value.